source/js/addons.js

```diff
@@ -0,0 +1,183 @@
+
+// For local development use thing-proxy with URL below or turn off CORS in browser
+// To use thing-proxy 'cd source/js/thing-proxy && npm i && node server.js'
+// REMEMBER to change it back before commit!
+// var apiUrl = "http://localhost:3000/fetch/https://api.playnite.link/api/addons"
+var apiUrl = "https://api.playnite.link/api/addons";
+
+var addons = [];
+var addonsGameLibrary = [];
+var addonsMetadataProvider = [];
+var addonsGeneric = [];
+var addonsThemeDesktop = [];
+var addonsThemeFullscreen = [];
+
+function sortByName(a, b) {
+    var aName = a.name.toLowerCase();
+    var bName = b.name.toLowerCase();
+    return ((aName < bName) ? -1 : ((aName > bName) ? 1 : 0));
+}
+
+$(document).ready(function () {
+    getAddonList();
+    hideAlerts();
+
+    $("#searchBox").keyup(function (data) {
+        if (data.target.value.length > 0) {
+            var searchResult = search(data.target.value);
+            resortAddons(searchResult.map(function (val) { return val.item }));
+            rerenderAddons();
+        }
+        else {
+            resortAddons(addons);
+            rerenderAddons();
+        }
+    });
+});
+
+function getAddonList() {
+    $.getJSON(apiUrl, function (data) {
+        addons = data.data.sort(sortByName);
+        resortAddons(data.data);
+        rerenderAddons();
+    });
+}
+
+function resortAddons(data) {
+    addonsGameLibrary = [];
+    addonsMetadataProvider = [];
+    addonsGeneric = [];
+    addonsThemeDesktop = [];
+    addonsThemeFullscreen = [];
+    $.each(data, function (_key, val) {
+        switch (val.type) {
+            case 1:
+                addonsMetadataProvider.push(val);
+                break;
+            case 2:
+                addonsGeneric.push(val);
+                break;
+            case 3:
+                addonsThemeDesktop.push(val);
+                break;
+            case 4:
+                addonsThemeFullscreen.push(val);
+                break;
+            case undefined:
+                addonsGameLibrary.push(val);
+                break;
+            default:
+        }
+    });
+}
+
+function rerenderAddons() {
+    renderAddonsSection(addonsMetadataProvider, "addonsMetadataProvider", "Metadata Sources");
+    renderAddonsSection(addonsGeneric, "addonsGeneric", "Generic");
+    renderAddonsSection(addonsThemeDesktop, "addonsThemeDesktop", "Themes Desktop");
+    renderAddonsSection(addonsThemeFullscreen, "addonsThemeFullscreen", "Themes Fullscreen");
+    renderAddonsSection(addonsGameLibrary, "addonsGameLibrary", "Libraries");
+}
+
+function sanitize(input) {
+    var output = input.replace(/<script[^>]*?>.*?<\/script>/gi, '').
+        replace(/<[\/\!]*?[^<>]*?>/gi, '').
+        replace(/<style[^>]*?>.*?<\/style>/gi, '').
+        replace(/<![\s\S]*?--[ \t\n\r]*>/gi, '').
+        replace(/&nbsp;/g, '');
+    return output;
```

Comment:
I wouldn't try to actually parse the HTML too much, just escape all the unsafe characters which are only these three:
```suggestion
    return input.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;");
```
This preserves the display of the characters in case someone legitimately wants to use them in their description too.

If your input already has HTML entities, escaping `&` can lead to double escaped `&amp;amp;`, but it sounds like this API is just the raw text from the YAML.

Using `encodeURI()` looks good for URLs, you don't need to worry about quotes here.